### PR TITLE
ns for fx: add basic testing on cuda

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65593

Summary:

Adds test cases that the three Numeric Suite Core APIs work
when the models are on cuda.  In particular:
1. create models and move them to cuda
2. add loggers (if applicable)
3. run data through (if applicable)
4. extract results

It works without code changes because a `Logger` object is
created without any device specific objects (they only get
added if a data is passed through). It's good to have this tested.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_extract_weights_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_loggers_cuda
python test/test_quantization.py TestFXNumericSuiteCoreAPIs.test_add_shadow_loggers_cuda
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D31160897](https://our.internmc.facebook.com/intern/diff/D31160897)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31160897).

**Review by raghuramank100 — APPROVED** [review]

**Review by hx89 — APPROVED** [review]

File: test/quantization/fx/test_numeric_suite_fx.py [hunk 1]
```diff
@@ -1854,6 +1855,54 @@ def test_shadow_loggers_preserve_qat_numerics(self):
         ref_shadow = mc_shadows_mp(datum)
         self.assertEqual(ref_fp32, ref_shadow)
 
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
```
Comment (jerryzh168) [inline test/quantization/fx/test_numeric_suite_fx.py:1858]: nit: do we need multigpu for this test?

**Review by jerryzh168 — APPROVED** [review]
lgtm

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31160897).

**Pull request closed without merging**